Sources/Rules/EmptyExtension.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  EmptyExtension.swift
+//  SwiftFormat
+//
+// Created by manny_lopez on 7/29/24.
+//  Copyright © 2024 Nick Lockwood. All rights reserved.
+//
+
+import Foundation
+
+public extension FormatRule {
+    /// Remove empty, non-conforming, extensions.
+    static let emptyExtension = FormatRule(
+        help: "Remove empty, non-conforming, extensions.",
+        orderAfter: [.unusedPrivateDeclaration]
+    ) { formatter in
+        var emptyExtensions = [Declaration]()
+
+        formatter.forEachRecursiveDeclaration { declaration in
+            let declarationModifiers = Set(declaration.modifiers)
+            guard declaration.keyword == "extension",
+                  let declarationBody = declaration.body,
+                  declarationBody.isEmpty,
+                  // Ensure that it is not a macro
+                  !declarationModifiers.contains { $0.first == "@" }
```

Comment:
I guess we also have to do:

```suggestion
                  !declarationModifiers.contains(where: { $0.first == "@" })
```